src/backend/distributed/deparser/qualify_text_search_stmts.c

```diff
@@ -69,6 +69,44 @@ QualifyDropTextSearchConfigurationStmt(Node *node)
 }
 
 
+/*
+ * QualifyDropTextSearchDictionaryStmt adds any missing schema names to text search
+ * dictionaries being dropped. All dictionaries are expected to exists before fully
+ * qualifying the statement. Errors will be raised for objects not existing. Non-existing
+ * objects are expected to not be distributed.
+ */
+void
+QualifyDropTextSearchDictionaryStmt(Node *node)
+{
+	DropStmt *stmt = castNode(DropStmt, node);
+	Assert(stmt->removeType == OBJECT_TSDICTIONARY);
+
+	List *qualifiedObjects = NIL;
+	List *objName = NIL;
+
+	foreach_ptr(objName, stmt->objects)
+	{
+		char *schemaName = NULL;
+		char *tsdictName = NULL;
+		DeconstructQualifiedName(objName, &schemaName, &tsdictName);
+
+		if (!schemaName)
+		{
+			Oid tsdictOid = get_ts_dict_oid(objName, false);
+			Oid namespaceOid = get_ts_dict_namespace(tsdictOid);
+			schemaName = get_namespace_name(namespaceOid);
+
+			objName = list_make2(makeString(schemaName),
+								 makeString(tsdictName));
```

Comment:
This is surprising that codecov thinks it is not covered.
Do you drop the Dicts always fully qualified? (I think I will figure that out once I read the tests :D)
Or do we have coverage reporting issues?

Reply:
> Do you drop the Dicts always fully qualified

Yes. I just thought that this was a reporting issue, but apparently it caught something here.

Updated the tests to have some drop operations on unqualified dictionaries.